scripts/react-native-xcode.sh

```diff
@@ -107,16 +107,11 @@ fi
 # shellcheck source=/dev/null
 source "$REACT_NATIVE_DIR/scripts/node-binary.sh"
 
-[ -z "$HERMES_PATH" ] && export HERMES_PATH="$REACT_NATIVE_DIR/../hermes-engine"
+[ -z "$HERMES_CLI_PATH" ] && HERMES_CLI_PATH="$PODS_ROOT/hermes-engine/destroot/bin/hermesc"
 
-export HERMES_CLI="$HERMES_PATH/osx-bin/hermesc"
-
-if [[ $USE_HERMES == true && ! -d "$HERMES_PATH" ]]; then
-  echo "error: Can't find Hermes executable - directory $HERMES_PATH doesn't exist. " \
-       "If you have a non-standard project structure, locate the 'Bundle React Native code and images' " \
-       "build phase in your Xcode project and set the `$HERMES_PATH` variable to the location of the " \
-       "hermes-engine package. For example: `$SRCROOT/../../node_modules/hermes-engine` " >&2
-  exit 2
+if [[ -z "$USE_HERMES" && -f "$HERMES_CLI_PATH" ]]; then
```

Comment:
Should we still have some of the above path validation when `USE_HERMES` is set to `true`?

Reply:
We could, I wonder what the messaging should be.

Theoretically, this will only happen IF `USE_HERMES` was set manually to `true`. Otherwise, our mechanism will never turn it on when file does not exist.

Shall the wording mention running `pod install` or just go directly into "setting a path to Hermes executable"?

Reply:
Hmm, good question. What about this?

> USE_HERMES is set to `true` but the `hermesc` binary could not be found at `${HERMES_CLI_PATH}`. Perhaps you need to run `pod install` or otherwise point the `HERMES_CLI_PATH` variable to your custom location.

(`${HERMES_CLI_PATH}` should be expanded to the actual path)

Reply:
Oh, amazing! Thank you! I don't feel super comfortable in wordings, this saves me a lot of time 🙏